src/main/java/com/fynd/extension/controllers/ExtensionController.java

```diff
@@ -59,25 +60,34 @@ public ResponseEntity install(@RequestParam(value = "company_id") String company
             if (ext.isOnlineAccessMode()) {
                 session = new Session(Session.generateSessionId(true, null), true);
             } else {
-                sid = Session.generateSessionId(false, new Option(companyId, ext.getCluster()));
+                sid = Session.generateSessionId(false, new Option(companyId,
+                                                                  ext.getExtensionProperties()
+                                                                     .getCluster()));
                 session = sessionStorage.getSession(sid);
                 if (ObjectUtils.isEmpty(session)) {
                     session = new Session(sid, true);
-                } else if (!Objects.equals(session.getExtension_id(), ext.getApi_key())) {
+                } else if (!Objects.equals(session.getExtension_id(),
+                                           ext.getExtensionProperties()
+                                              .getApi_key())) {
                     session = new Session(sid, true);
                 }
             }
 
-            Date sessionExpires = Date.from(Instant.now().plusMillis(900000));
+            Date sessionExpires = Date.from(Instant.now()
+                                                   .plusMillis(900000));
```

Comment:
Use this value from property file. Check @refreshscope in Spring boot

Reply:
This is not a web based application with any endpoints. Has no SpringBoot application context , therefore we cannot introduce yaml file